client/my-sites/checklist/wpcom-checklist/component.jsx

```diff
@@ -0,0 +1,567 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import page from 'page';
+import PropTypes from 'prop-types';
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import { compact, find, get, some } from 'lodash';
+import { isDesktop } from 'lib/viewport';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { getTaskList } from './wpcom-task-list';
+import Checklist from 'components/checklist';
+import ChecklistBanner from './checklist-banner';
+import ChecklistBannerTask from './checklist-banner/task';
+import ChecklistNavigation from './checklist-navigation';
+import ChecklistNotification from './checklist-notification';
+import getSiteChecklist from 'state/selectors/get-site-checklist';
+import isEligibleForDotcomChecklist from 'state/selectors/is-eligible-for-dotcom-checklist';
+import QueryPosts from 'components/data/query-posts';
+import QuerySiteChecklist from 'components/data/query-site-checklist';
+import Task from 'components/checklist/task';
+import { createNotice } from 'state/notices/actions';
+import { getPostsForQuery } from 'state/posts/selectors';
+import { getSelectedSiteId, isSiteSection } from 'state/ui/selectors';
+import { getSiteOption, getSiteSlug } from 'state/sites/selectors';
+import { recordTracksEvent } from 'state/analytics/actions';
+import { requestGuidedTour } from 'state/ui/guided-tours/actions';
+import { requestSiteChecklistTaskUpdate } from 'state/checklist/actions';
+import { getCurrentUser, isCurrentUserEmailVerified } from 'state/current-user/selectors';
+import userFactory from 'lib/user';
+import { launchSite } from 'state/sites/launch/actions';
+import isUnlaunchedSite from 'state/selectors/is-unlaunched-site';
+import createSelector from 'lib/create-selector';
+
+const userLib = userFactory();
+
+const query = { type: 'any', number: 10, order_by: 'ID', order: 'ASC' };
```

Comment:
This could be a constant with a nice uppercase name: `FIRST_TEN_SITE_POSTS_QUERY`.

Reply:
Done in #28678.